run.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-cd `dirname $0`
+cd "$(dirname "$0")"
```

Comment:
POSIX unfortunately doesn't specify that `/bin/sh` be POSIX-compliant, just that a POSIX-compliant `sh` (which supports `$(...)`) be installed. On Solaris this was (for a long time) `/usr/xpg4/bin/sh`, on newer Solaris I don't know what the case is.

In the past, having come from the Solaris world and having felt the pain of `sh`-shebanged scripts using non-Bourne syntax, I always wrote my `sh`-shebanged scripts to strict Bourne syntax. I am less convinced that that is necessary in 2016, however. Are there still modern POSIX operating systems for which `/bin/sh` is not POSIX 1003.2/2a compliant?